qiskit/assembler/disassemble.py

```diff
@@ -40,6 +40,14 @@
 def disassemble(qobj) -> Union[CircuitModule, PulseModule]:
     """Disassemble a qobj and return the circuits or pulse schedules, run_config, and user header.
 
+    .. note::
+
+        ``disassemble(assemble(qc))`` is not guaranteed to produce an exactly equal circuit to the
+        input, due to limitations in the :obj:`.QasmQobj` format that need to be maintained for
```

Comment:
`` :obj:`.x` `` only displays what's after the `.` (I think), and in module/class `a.b.C` it has the search path:
1. `a.b.C.x`
2. `a.b.x`
3. `a.x`
4. `x`
5. _any prefix_`.x`
Sphinx is meant to warn if there are duplicate names that it could resolve to, but I've never actually tried that myself - if I know there are potentially duplicate names, I tend to use slightly longer forms because if it's confusing for me/Sphinx, it'll probably be confusing for the reader too.  For example, in the release notes I used `` :meth:`.InstructionSet.c_if` `` to avoid conflict with `Instruction.c_if`, and that displays like this:

<img width="795" alt="Screenshot 2021-11-16 at 10 05 15" src="https://user-images.githubusercontent.com/5968590/141964989-91b6a17a-8435-46be-bb24-faae51ac997f.png">